frontend/js/src/explore/fresh-releases/components/ReleaseTimeline.tsx

```diff
@@ -88,6 +151,16 @@ function createMarks(
       .map((_, index, arr) =>
         arr.slice(0, index + 1).reduce((prev, curr) => prev + curr)
       );
+
+    if (sortDirection === "descend") {
+      dataArr.reverse();
+      percentArr = percentArr.reverse().map((v) => (v <= 100 ? 100 - v : 0));
+    }
+
+    if (percentArr[0] !== 0) {
+      percentArr.unshift(0);
+      percentArr.pop();
+    }
```

Comment:
This block of code is now repeated multiple times, which is not great.

Since it applies in all cases apart from order === "release_date", you can simply apply it conditionally at the end of the function, after all the other blocks.

```
if (order !== "release_date") {
    if (sortDirection === "descend") {
      dataArr.reverse();
      percentArr = percentArr.reverse().map((v) => (v <= 100 ? 100 - v : 0));
    }

    if (percentArr[0] !== 0) {
      percentArr.unshift(0);
      percentArr.pop();
    }
  }
```